active_plugins/rundeepprofiler.py

```diff
@@ -0,0 +1,375 @@
+        shutil.copy(f"{model_dir}\{self.model_filename.value}", f"{out_directory}\outputs\{self.assay.value}\checkpoint")
+        shutil.copy(f"{config_dir}\{self.config_filename.value}", f"{out_directory}\inputs\config")
+
+        #
+        # Locations file
+        # creates a location file for each site with the nuclei_center
+        #
+        # Get the measurements object
+        measurements = workspace.measurements
+        # Get inputs
+        x_obj = measurements.get_current_measurement(self.input_object_name.value, "Location_Center_X")
+        y_obj = measurements.get_current_measurement(self.input_object_name.value, "Location_Center_Y")
+        plate = measurements.get_current_image_measurement(self.metadata_plate.value)
+        well = measurements.get_current_image_measurement(self.metadata_well.value)
+        site = measurements.get_current_image_measurement(self.metadata_site.value)
+        print(plate, well, site)
+        # Create plate directory and location file
+        loc_dir = f"{out_directory}\inputs\locations\{self.metadata_plate.value}"
+        loc_file = f"{well}-{site}-Nuclei.csv"
+        os.makedirs(f"{out_directory}\inputs\locations\{self.metadata_plate.value}", exist_ok=True)
+        # Create the actual file
+        header = 'Nuclei_Location_Center_X', 'Nuclei_Location_Center_Y'
+        csvpath = os.path.join(loc_dir, loc_file)
+        with open(csvpath, 'w', newline='', encoding='utf-8') as fpointer:
+                writer = csv.writer(fpointer)
+                writer.writerow(header)
+                for i in range(len(x_obj)):
+                    writer.writerow((x_obj[i], y_obj[i]))
+        #
+        # Create the index/metadata file
+        #
+        header_files = ["Metadata_Plate", "Metadata_Well", "Metadata_Site"]
+        filename_list = [self.metadata_plate.value, well, site]
+        for img in self.images_list.value:
+            pathname = measurements.get_current_image_measurement(f"PathName_{img}")
+            filename = measurements.get_current_image_measurement(f"FileName_{img}")
+            filename_list.append(f"{pathname}\{filename}")
+        header_files.extend(self.images_list.value)
+        print(filename_list)
+        print(header_files)
+        index_dir = f"{out_directory}\inputs\metadata"
+        index_file = f"index_{str(random.randint(100000, 999999))}.csv"
+        indexpath = os.path.join(index_dir, index_file)
+        with open(indexpath, 'w', newline='', encoding='utf-8') as fpointer:
+                writer = csv.writer(fpointer)
+                writer.writerow(header_files)
+                writer.writerow((filename_list))
+              
+        #
+        # RUN!
+        #
+        cmd_run = f"python {executable} --root={out_directory} --config {self.config_filename.value} --metadata {index_file} --exp {self.assay.value} profile"
+        subprocess.run(cmd_run, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
```

Comment:
Right now, we're not capturing if we run successfully or not. I think something like this would be good
```
        result = subprocess.run(cmd_run.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
        if result.returncode!=0:
            msg = f"The call to DeepProfiler returned an error; you should try to solve this outside of CellProfiler. DeepProfiler output waws as follows: \n {result.stdout}"
            raise RuntimeError(msg)
```